Tests/BasicsTests/Archiver/ZipArchiverTests.swift

```diff
@@ -256,18 +256,18 @@ class ArchiverTests: XCTestCase {
 }
 
 extension Archiver {
-    func extract(from: AbsolutePath, to: AbsolutePath) throws {
-        try temp_await {
+    func extract(from: AbsolutePath, to: AbsolutePath) async throws {
+        try await safe_async {
             self.extract(from: from, to: to, completion: $0)
         }
     }
-    func compress(directory: AbsolutePath, to: AbsolutePath) throws {
-        try temp_await {
+    func compress(directory: AbsolutePath, to: AbsolutePath) async throws {
+        try await safe_async {
             self.compress(directory: directory, to: to, completion: $0)
         }
     }
-    func validate(path: AbsolutePath) throws -> Bool {
-        try temp_await {
+    func validate(path: AbsolutePath) async throws -> Bool {
```

Comment:
For these two, why not just add them to `Sources/Basics/Archiver/Archiver.swift` directly? But also they probably shouldn't use `safe_async` with a `DispatchQueue` under the hood, but `try await withCheckedThrowingContinuation`. Ideally when full conversion to Swift Concurrency is done, we don't have any mentions of Dispatch in the codebase and we let the concurrency thread pool manage scheduling instead of Dispatch.

Reply:
I don't think we can directly use withCheckedThrowingContinuation safely because there is no guarantee that a method with callback is actually non-blocking. It could be implemented like this:

```
func doSomething(_ complete: () -> Void) {
  semaphore.wait()
  complete()
}
```

Because the block of the withCheckedThrowingContinuation runs on the swift concurrency pool to safely transition to async you have to push off the concurrency pool to make sure you are not blocking a concurrency pool thread.

Reply:
Specifically archivers code doesn't use semaphores under the hood. In fact, it offloads its work with `DispatchQueue.sharedConcurrent`, so wrapping archivers in `safe_async` feels redundant to me.

Reply:
Yeah I can defer adding the safe_async method until it is for sure needed, but I figured that it provides protection against missing a un unexpected block or someone adding a blocking operation later